### PR TITLE
Hot Module Replacement (HMR) Integration

### PR DESCRIPTION
**Pull request by minowau**
solving issue #1255

This project implements Hot Module Replacement (HMR) for a React-based Chrome extension, allowing developers to see real-time changes in their popup component without needing to refresh the entire extension. The following outlines the configuration and structure necessary for achieving this setup.

#### Key Components

1. **Webpack Configuration (`webpack.config.js`)**:
   - Configures Webpack for development and production modes, specifying the entry point and output settings.
   - Enables HMR in the development server for instant updates.

2. **Entry Point (`src/popup.js`)**:
   - Contains the main logic for the popup component, integrating the HMR logic to allow for module updates without a full reload.

3. **Index File (`src/index.js`)**:
   - Updated to support HMR with checks for `module.hot`, ensuring that the component re-renders on updates without refreshing the entire popup.

4. **Package Configuration (`package.json`)**:
   - Includes scripts for building and serving the application, specifying configurations needed for both development and production.

5. **Development Server**:
   - The command `npm start` launches a development server with HMR enabled, providing a smooth development experience.

#### Summary of Changes

- **HMR Logic**:
  - Added in `popup.js` using `if (module.hot) { ... }` to ensure updates are reflected in real-time.
  - Implemented in `index.js` to facilitate automatic re-rendering of the popup component on code changes.

- **Webpack Dev Server**: Configured with `hot: true` to support HMR functionality.

- **File Structure**: Organized files into a clear structure, facilitating maintainability and ease of access.

### Benefits

Implementing HMR improves the development workflow by reducing the time spent on refreshing and waiting for the extension to reload. This results in a more productive environment, allowing for faster iteration and debugging of features.

# ✨ Pull Request

### CODE REVIEW

**Comment by minowau** [comment]
@manojVivek I hope this would solve your doubts.And you can check my code as well.I hope u star it

File: desktop-app-legacy/app/components/LiveCssEditor/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import React, {useMemo, useState, useEffect, useRef} from 'react';
-import {Rnd} from 'react-rnd';
-import {useSelector, useDispatch} from 'react-redux';
+import React, { useMemo, useState, useEffect, useRef } from 'react';
```
Comment (violetadev) [inline desktop-app-legacy/app/components/LiveCssEditor/index.js:1]: what's the purpose of these changes in the legacy app?

**Comment by violetadev** [comment]
I tested it and the reloading is faster when you change any css :)

File: desktop-app-legacy/app/components/LiveCssEditor/index.js (hunk 1 above)
Comment (minowau) [reply desktop-app-legacy/app/components/LiveCssEditor/index.js:1]: This would give an simplified version

Comment (minowau) [reply desktop-app-legacy/app/components/LiveCssEditor/index.js:1]: Better understanding

**Comment by minowau** [comment]
Can U merge the pull request

File: desktop-app-legacy/app/components/LiveCssEditor/index.js (hunk 1 above)
Comment (manojVivek) [reply desktop-app-legacy/app/components/LiveCssEditor/index.js:1]: @minowau We don't actively do any development on the legacy app. So, for the sake of clean git history, can you please revert these changes so that we can merge the PR?

**Comment by violetadev** [comment]
@minowau in your branch you can do:
`git checkout origin/main path-to-file` and it will make that file be the same as main. Then you add it to a new commit normally and you’re good to go

**Comment by minowau** [comment]
oh thanks for the help I forgot

**Comment by minowau** [comment]
I tried it but It 

> @minowau in your branch you can do: `git checkout origin/main path-to-file` and it will make that file be the same as main. Then you add it to a new commit normally and you’re good to go

I tried but I was not able to do it. If U can help U can have my credentials and U can try or U CAN JUST MERGE IT and change it afterwards with an other pull it is just a one line code

**Comment by violetadev** [comment]
> I tried it but It
> 
> > @minowau in your branch you can do: `git checkout origin/main path-to-file` and it will make that file be the same as main. Then you add it to a new commit normally and you’re good to go
> 
> I tried but I was not able to do it. If U can help U can have my credentials and U can try or U CAN JUST MERGE IT and change it afterwards with an other pull it is just a one line code

don't give your credentials to anyone please. Can you do from the root of the project `git checkout origin/main desktop-app-legacy/app/components/LiveCssEditor/index.js` ? what does it say when you do that?

**Comment by minowau** [comment]
it just gives the code a better look but no changes were done withe the change in the location of words in the brackets.Sorry but I tried but it didn't work on my side

**Comment by minowau** [comment]
can u add me as a contibutor

**Comment by violetadev** [comment]
@all-contributors Please add @minowau for code.
